src/accelerate/commands/estimate.py

```diff
@@ -0,0 +1,206 @@
+
+    Args:
+        model_name (`str`):
+            The model name on the Hub
+        library_name (`str`):
+            The library the model has an integration with, such as `transformers`. Will be used if `model_name` has no
+            metadata on the Hub to determine the library.
+        trust_remote_code (`bool`, `optional`, defaults to `False`):
+            Whether or not to allow for custom models defined on the Hub in their own modeling files. This option
+            should only be set to `True` for repositories you trust and in which you have read the code, as it will
+            execute code present on the Hub on your local machine.
+
+    """
+    model_info = verify_on_hub(model_name)
+    if not model_info:
+        raise ValueError(f"Model `{model_name}` is not an available model on the Hub.")
+    if library_name is None:
+        library_name = getattr(model_info, "library_name", False)
+        if not library_name:
+            raise ValueError(
+                f"Model `{model_name}` does not have any library metadata on the Hub, please manually pass in a `--library_name` to use (such as `transformers`)"
+            )
+    if library_name == "transformers":
+        if not is_transformers_available():
+            raise ImportError(
+                f"To check `{model_name}`, `transformers` must be installed. Please install it via `pip install transformers`"
+            )
+        print(f"Loading pretrained config for `{model_name}` from `transformers`...")
+        # As we just load in the `config` file and not the weights, we use `True`
+        has_remote_code = hasattr(model_info.config, "auto_map")
+        if has_remote_code and not trust_remote_code:
+            raise ValueError(
+                f"Loading {model_name} requires to execute some code in that repo, you can inspect the content of "
+                f"the repository at https://hf.co/{model_name}. You can bypass this by passing "
+                "`--trust_remote_code`"
+            )
+        config = AutoConfig.from_pretrained(model_name, trust_remote_code=trust_remote_code)
+        with init_empty_weights():
+            model = AutoModel.from_config(config, trust_remote_code=trust_remote_code)
+    elif library_name == "timm":
+        if not is_timm_available():
+            raise ImportError(
+                f"To check `{model_name}`, `timm` must be installed. Please install it via `pip install timm`"
+            )
+        print(f"Loading pretrained config for `{model_name}` from `timm`...")
+        with init_empty_weights():
+            model = timm.create_model(model_name, pretrained=False)
+    else:
+        raise ValueError(
+            f"Library `{library_name}` is not supported yet, please open an issue on GitHub for us to add support."
+        )
+    return model
+
+
+def create_ascii_table(headers: list, rows: list, title: str):
+    "Creates a pretty table from a list of rows, minimal version of `tabulate`."
```

Comment:
I would have just used tabulate. It has extremely minimal requirements and has never once caused issues in years of using it.

Reply:
While it is, we tend to avoid adding any dependencies to accelerate whatsoever, and here it was a small enough code chunk to work fine. E.g. see the CLI menu selector